test/hotspot/jtreg/runtime/ErrorHandling/CreateCoredumpOnCrash.java

```diff
@@ -36,31 +36,72 @@
 import jdk.test.lib.Platform;
 import jdk.internal.misc.Unsafe;
 
+import java.lang.Runtime;
+
 public class CreateCoredumpOnCrash {
-    private static class Crasher {
-        public static void main(String[] args) {
-            Unsafe.getUnsafe().putInt(0L, 0);
-        }
+  private static class Crasher {
+    public static void main(String[] args) {
+      Unsafe.getUnsafe().putInt(0L, 0);
+    }
+  }
+
+  private static String ulimitString(int limit) {
+    String string = "ulimit -c ";
+    if (limit != Integer.MAX_VALUE) {
+      string += limit;
+    } else {
+      string += "unlimited";
     }
+    return string+";";
+  }
 
-    public static void main(String[] args) throws Exception {
-        runTest("-XX:-CreateCoredumpOnCrash").shouldContain("CreateCoredumpOnCrash turned off, no core file dumped")
-                                             .shouldNotHaveExitValue(0);
+  public static void main(String[] args) throws Exception {
+    runTest("-XX:-CreateCoredumpOnCrash").shouldContain("CreateCoredumpOnCrash turned off, no core file dumped")
+    .shouldNotHaveExitValue(0);
 
-        if (Platform.isWindows()) {
-            // The old CreateMinidumpOnCrash option should still work
-            runTest("-XX:-CreateMinidumpOnCrash").shouldContain("CreateCoredumpOnCrash turned off, no core file dumped")
-                                                 .shouldNotHaveExitValue(0);
-        } else {
-            runTest("-XX:+CreateCoredumpOnCrash").shouldNotContain("CreateCoredumpOnCrash turned off, no core file dumped")
-                                                 .shouldNotHaveExitValue(0);
-        }
+    if (Platform.isWindows()) {
+      // The old CreateMinidumpOnCrash option should still work
+      runTest("-XX:-CreateMinidumpOnCrash").shouldContain("CreateCoredumpOnCrash turned off, no core file dumped")
+      .shouldNotHaveExitValue(0);
+    } else {
+      OutputAnalyzer oa = new OutputAnalyzer(Runtime.getRuntime().exec("ulimit -c"));
+      oa.shouldHaveExitValue(0);
+      if (!oa.contains("0\n")) {
+        oa = runTest("-XX:+CreateCoredumpOnCrash");
+        oa.shouldContain("Core dump will be written.");
+        oa.shouldNotHaveExitValue(0);
 
+        oa = runTest("-XX:+CreateCoredumpOnCrash", ulimitString(1024));
+        oa.shouldContain("warning: CreateCoredumpOnCrash specified, but");
+        oa.shouldContain("To ensure a full core dump");
+        oa.shouldNotHaveExitValue(0);
+
+        // Do "ulimit -c 0" last, as once we decrease it, we cannot increase it
```

Comment:
But you are running each test in a new shell.

Reply:
Good catch. The comment is a leftover from earlier prototypes.

I will remove it.